### ISSUE TITLE
Over zooming of ui VST3

### ISSUE THREAD

**Issue by hfiguiere** [surge-synthesizer/surge#7396]
**Bug Description:**
Surge over zoom the UI in Ardour 8.

**Surge XT Version**
1.2.3 or 1.3.0

* Version:
* Plugin Type: VST3
* Bitness: x86_64

**Reproduction Steps:**
1. Start Ardour
2. In Ardour preference set a font size to 125%. (it's still too small on my 4K screen)
3. Create track in Ardour to use Surge XT VST3
4. Open the UI. It's too big.

**Expected Behavior:**
The UI is properly sized.

**Screenshots:**
![Screenshot from 2023-12-13 16-16-58](https://github.com/surge-synthesizer/surge/assets/114441/95a04607-f530-4ab2-9d76-b2478a8ddf7a)

**Computer Information (please complete the following!):**
 - OS: Linux Fedora
 - Host: Ardour 8.1 or 8.2
 - Version: 

**Additional Information:**
Source: https://github.com/flathub/org.surge_synth_team.surge-xt/pull/15#issuecomment-1854755320

tl;dr Ardour give the UI scale to the plugin. And the plugins seenm to do it twice.
I can't even change the ui zoom in Surge.
Also 150%, I get the black window.

https://github.com/flathub/org.surge_synth_team.surge-xt/pull/15#issuecomment-1854774369

**Comment by hfiguiere** [comment]
Just to clarify, the above screenshot is NOT edited. The crop was taken as is.

**Comment by mkruselj** [comment]
We think this is an issue on Ardour end, as we don't see the same behavior in any other VST3 host.

**Comment by miranda** [comment]
I just installed Surge XT on Arch Linux, using it with Ardour 8... discovered the same thing.  It's so huge it won't fit on my screen.  But I also noticed none of the zoom or shrink settings do anything at all, even trying to enlarge it.  I also noticed that it has the option "Set Current Zoom Level (250%) as Default" and "Set Default Zoom Level to..." brings up 250 to start with, and setting it to a different number won't change it.  So it's obviously stuck at 250% zoom.

Another thing that it's doing is when I bring up the menu and move the mouse between selections, the Surge UI completely disappears and only the menu is visible.  Not sure if that is related.

**Comment by mkruselj** [comment]
I really really think this is a bug in Ardour...

**Comment by erikjms** [comment]
Might this bug be related?

[Ardour: VST3 Plug-In GUIs cannot be resized smaller than the size at the most recent project save](https://tracker.ardour.org/view.php?id=8764)

It's the most recent bug I can find that looks at all relevant on Ardour's bug tracker. It's been open since 2021. I have not tried the workarounds suggested but will. 

In the meantime, setting Ardour GUI scale back to 100% and restarting makes Surge's GUI zoomable again. Tweaking presets in tiny Ardour and then using them in readable Ardour is somewhat cumbersome, but since they both do exactly what I want otherwise, that is what I have been doing.

I'd be grateful if you all could talk to the Ardour devs about this when you have a chance. I do not have the resources to argue-- or, um, _discuss_ this with them. Making noise/music is much more life-affirming. :)

**Comment by mkruselj** [comment]
I suppose this means we could probably close this issue if it's confirmed on Ardour's end (which was my suspicion from the get go).

**Comment by mkruselj** [comment]
@baconpaul Thoughts on the above?

**Comment by baconpaul** [comment]
I was going to test it next time i was in linux. Lets leave it be open for now on the odd chance that up upgrade to juce 7.0.12 for #7640 also helps

**Comment by atomikaya** [comment]
I found a curious workaround: if I open Surge as a standalone, set the UI to 100% and save as default, then restart Ardour... Voilà :ok_hand: My plugin opens at a 100% (still a lil bit scaled up from the Ardour 125% setting but that's an Ardour issue probably).

Not a real fix but I hope it helps.

Fedora 40 / flatpak Ardour 8.6.0 / flatpak Surge XT 1.3.4 as VST3 / held together with a good helping of Flatseal tape / a Keystation Mini 32 to make the beep boops

**Comment by mkruselj** [comment]
@baconpaul What do we do here?

**Comment by baconpaul** [comment]
I just updated to juce 8.0.4

I think i just give it a whirl when I'm in linux for other 1.4.0 issues and see what we can do.

My guess is ardour is mis-sending scaling information via the vst3 api and we are obeying it more strictly than others. but that's just a guess

**Comment by petter1** [comment]
I found this:

https://steinbergmedia.github.io/vst3_dev_portal/pages/Technical+Documentation/Change+History/3.6.6/IPlugViewContentScaleSupport.html

@baconpaul did you find out where the default is stored? I do not have standalone and want try if I can workaround by changing a file

**Comment by baconpaul** [comment]
Yes the host tells the plugin the scale using that api in vst3 (And a similar one in clap)

sure sounds like arduour is sending it in a way which is applied twice

I have not debugged it at all but yes now you see my point!

**Comment by fred-o** [comment]
FWIW, I stumbled over a simple workaround for this issue. Open the file `SurgeXTUserDefaults.xml` (should be in your `~/Documents/Surge XT` folder on Linux) in a text editor and add the line 
```
<default key="defaultZoom" value="100" type="2" />
```
to the other defaults. Save and restart Ardour. After that I was able to resize Surge XT using the Zoom menu as normal.

**Comment by DevJac** [comment]
In Ardour if I decrease the "GUI and Font scaling", which is located in Edit -> Preferences -> Appearance -> Size and Scale --

If I lower the "GUI and Font scaling" in Ardour, then it will open Surge XT at a smaller size that can fit on my screen.

I agree this is an Ardour issue.

I suspect that Ardour will not allow the VST screen to shrink. I notice this because I can repeatedly tell Surge XT to shrink, and it does appear to shrink for a fraction of a second, and then it goes back to the previous size. I suspect there's some logic in Ardour that refuses to allow a VST window to be smaller than its initial size.

**Comment by DevJac** [comment]
This should be fixed in Ardour.

If anyone is having this issue, try right clicking the background of Surge XT somewhere, and it should bring up a menu with some zoom options. Lower the zoom until you can see the whole plugin.

Ardour used to not allow plugins to shrink smaller than their initial size, but that should be fixed now, and so this issue should be resolved. Ardour versions after 9.8 should have this fix.
